docs/manuscripts/cosmicqc_paper_v1/figure_3/nbconverted/bootstrap_silho_scores.py

```diff
@@ -132,14 +135,69 @@ def bootstrap_silhouette(  # noqa: PLR0913
 print("Silhouette score (excluding noise):", pre_QC_score)
 
 
+# # Output which cells go with which cluster (plot bar plot proportion of cells removed on y and the x is the cluster label)
+#
+# Is there a way to show that the cluster cells that did not get removed moved to a different cluster?
+
 # In[5]:
 
 
+import matplotlib.pyplot as plt
+
+# Attach cluster labels to full dataframe
+pre_QC_clusters_df = pre_QC_umap_df.copy()
+pre_QC_clusters_df["cluster"] = pre_cluster_labels
+
+# Drop noise
+pre_QC_clusters_df = pre_QC_clusters_df[pre_QC_clusters_df["cluster"] != -1]
+
+# Convert labels to string for nicer plotting
+pre_QC_clusters_df["cluster"] = pre_QC_clusters_df["cluster"].astype(str)
+
+# Print number of cells in each cluster
+print(pre_QC_clusters_df["cluster"].value_counts().sort_index())
+
+# Calculate proportions for each cluster
+cluster_counts = (
+    pre_QC_clusters_df.groupby(["cluster", "Metadata_QC_status"])
+    .size()
+    .unstack(fill_value=0)
+)
+cluster_props = cluster_counts.div(cluster_counts.sum(axis=1), axis=0)
+
+# Plot
+ax = cluster_props.plot(
```

Comment:
In addition to what you created, would it also make sense to visualize the clusters of the points? I wonder if this could provide a further sense of validation on what we're seeing from this plot.

Reply:
This is from the UMAP so we do have that available :)